src/main/scala/com.snowplowanalytics.snowplow/scalatracker/TrackerExceptionHandler.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2015-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.scalatracker
+
+import java.io.{PrintWriter, StringWriter}
+
+import org.json4s.JsonDSL._
+
+import scala.util.control.NonFatal
+
+object TrackerExceptionHandler {
+
+  /**
+    * ExceptionHandler class
+    *
+    * @param tracker Tracker
+    */
+  implicit class ExceptionHandler(tracker: Tracker) {
```

Comment:
I'm not sure we should use `implicit class` here. For me it should be usual method in `Tracker`, probably called `trackError` for consistency with other trackers.

Reply:
Well `trackError` inside Tracker is a possibility. The implicit class here is complementing the behaviour of tracker. I've created this way to split the behaviour in small components, then I think it is easier to test and to maintain. However, if you prefer I can refactoring to `trackError 
 method`. What do you think?

Reply:
Yes, please, I think refactoring it is a good idea. Feel free to have `object ErrorTracking` with corresponding `SchemaKey` or other functions, but main method I believe should live in `Tracker`.